packages/mathjax-extension/src/index.ts

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+/**
+ * @packageDocumentation
+ * @module mathjax-extension
+ */
+
+import {
+  JupyterFrontEnd,
+  JupyterFrontEndPlugin
+} from '@jupyterlab/application';
+
+import { ILatexTypesetter } from '@jupyterlab/rendermime';
+
+// MathJax core
+import { mathjax } from 'mathjax-full/js/mathjax';
+
+// TeX input
+import { TeX } from 'mathjax-full/js/input/tex';
+
+// HTML output
+import { CHTML } from 'mathjax-full/js/output/chtml';
+
+import { TeXFont } from 'mathjax-full/js/output/chtml/fonts/tex';
+
+import { AllPackages } from 'mathjax-full/js/input/tex/AllPackages';
+
+import { SafeHandler } from 'mathjax-full/js/ui/safe/SafeHandler';
+
+import { HTMLHandler } from 'mathjax-full/js/handlers/html/HTMLHandler';
+
+import { browserAdaptor } from 'mathjax-full/js/adaptors/browserAdaptor';
```

Comment:
It's not clear to me whether this is possible, since there is a top-level statement that requires `mathjax` to be imported:

```
mathjax.handlers.register(SafeHandler(new HTMLHandler(browserAdaptor())));
```

If this can be done lazily when `typeset()` is called, then yes; otherwise no.

Reply:
I think you can call it later, just define a global variable to track of it was already called.

There is a trade-off with importing it only at first typeset call. An alternative would be to import async on application ready signal, this way we still get the saving for first load but without the penalty when first opening a notebook with math.

Reply:
Implemented in latest revision.